core/frontend/src/components/wifi/WifiUpdater.vue

```diff
@@ -0,0 +1,129 @@
+        timeout: 1000,
+      })
+        .then((response: AxiosResponse) => {
+          common_store.closeError(ErrorIndex.WifiStatusFail)
+
+          wifi_store.setNetworkStatus(response.data)
+
+          if (response.data.wpa_state != 'COMPLETED') {
+            wifi_store.setCurrentNetwork(null)
+            return
+          }
+
+          const scanned_network = wifi_store.available_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+          const saved_network = wifi_store.saved_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+
+          wifi_store.setCurrentNetwork({
+            ssid: response.data.ssid,
+            signal: scanned_network ? scanned_network.signal : 0,
+            locked: response.data.key_mgmt.includes('WPA'),
+            saved: saved_network ? true : false,
+          })
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiStatusFail,
+            message: `Could not fetch wifi status: ${error.message}`,
+          })
+          wifi_store.setCurrentNetwork(null)
+        })
+    },
+    async fetchAvailableNetworks (): Promise<void> {
+      await axios({
+        method: 'get',
+        url: `${wifi_store.API_URL}/scan`,
+        timeout: 15000,
+      })
+        .then((response) => {
+          common_store.closeError(ErrorIndex.WifiScanFail)
+
+          const saved_networks_ssids = wifi_store.saved_networks.map((network: SavedNetwork) => network['ssid'])
+          const available_networks = response.data.map((network: WPANetwork) => {
+            return {
+              ssid: network.ssid,
+              signal: network.signallevel,
+              locked: network.flags.includes('WPA'),
+              saved: saved_networks_ssids.includes(network.ssid),
+            }
+          })
+          wifi_store.setAvailableNetworks(available_networks)
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiScanFail,
+            message: `Could not scan wifi networks: ${error.message}`,
+          })
+          wifi_store.setAvailableNetworks([])
+        })
+    },
+    async fetchSavedNetworks (): Promise<void> {
+      await axios({
+        method: 'get',
+        url: `${wifi_store.API_URL}/saved`,
+        timeout: 1000,
+      })
+        .then((response) => {
+          common_store.closeError(ErrorIndex.WifiSavedFail)
+          wifi_store.setSavedNetworks(response.data)
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiSavedFail,
+            message: `Could not fetch saved networks: ${error.message}.`,
+          })
+          wifi_store.setSavedNetworks([])
+        })
+    },
+    sleep (ms: number): Promise<void> {
+      return new Promise((resolve) => setTimeout(resolve, ms))
+    },
+    async updateNetworkStatus (): Promise<void> {
```

Comment:
are these loops that run every 3s? There must be a better way to do this. 

wouldn't 
```
while (true) {
     await this.doStuff()
     await this.sleep(3000)
}
``` 
yield the same result, but in a more obvious way?

Reply:
Better, indeed. Changing.

Reply:
Will, in fact this approach will lock javascript in the while block. Do you know some better alternative considering the async/await approach?